projects/views.py

```diff
@@ -81,6 +85,23 @@ def user_projects(request):
     return render(request, "projects/user_projects.html", context)
 
 
+@login_required
+def accept_invite(request, invite_code):
+    mapper = ProjectAllocationMapper(request)
+    try:
+        accepted, result = mapper.accept_invite(request.user, invite_code)
```

Comment:
Some context: this `ProjectAllocationMapper` was more or less an implementation of the [Strangler pattern](https://martinfowler.com/bliki/StranglerFigApplication.html). We introduced it because we were in a situation where we had to simultaneously support old legacy users and new federated users, but we had tons of code doing common operations against each type of user, and those operations had very different behavior depending on this.

We have since migrated over finally, so we should start trying to tear down some of the complexity in this mapper. For this reason I don't think we should be adding more logic here.
I think it's arguably simpler to just do:

```python
invite = Invitation.get(invite_code=invite_code)  # or similar... see other comments about using ID instead
accepted = invite.accept(request.user)
if accepted:
    # message...
    return HttpResponseRedirect(reverse("projects:view_project", args=[invite.project])
else:
    # message...
```

Reply:
Edit: I saw that accept_invite is doing more stuff, notably actually adding the project membership. What if you _only_ used the mapper to add the project membership, but everything else was done here, or in a helper function in this file? That would keep the mapper's use minimized and make it easier to refactor later.